signer/tests/integration/setup.rs

```diff
@@ -457,9 +455,6 @@ pub async fn backfill_bitcoin_blocks(db: &PgStore, rpc: &Client, chain_tip: &bit
         db.write_bitcoin_block(&bitcoin_block).await.unwrap();
         block_header = rpc.get_block_header_info(&parent_header_hash).unwrap();
     }
-
-    let block_hash = db.get_bitcoin_canonical_chain_tip().await.unwrap().unwrap();
-    assert_eq!(block_hash.deref(), chain_tip);
```

Comment:
qq: why this deletion? If we are backfilling in a test from a tip and end up with a fork as canonical chain tip, there's probably something strange with the setup (or we risk having gaps in the chain, if we have no forks but backfilled from not a chain tip but we have a different tip in the db).

Reply:
I think the issue is that this function doesn't care about whether the chain is canonical or whether it is actually at the tip of any particular block chain. This function backfills from the given block hash to some minimum height, it doesn't really care about anything else. Since the given block hash need not be the actual chain tip,  the chain tip in the database does not need to match it, and that is the problem with the check. 

So, I should probably move the check to the `fetch_canonical_bitcoin_blockchain` helper, but it shouldn't be here. I should also rename the variable, I forgot to come back and do that.

Reply:
In general I agree -- this function does the backfill from a block, why should it care about what's in the db -- but without the check we risk ending up with gaps in the db view of the chain, which I think could be more annoying than just fixing the test setup to avoid triggering this assert. But I may be biased because I didn't encounter any test scenario where this check was triggering, so I may be missing some good reasons to remove this (other than the general one).

Reply:
Oh I forgot to change some things, one sec.

Reply:
Oh apparently I made one of the changes that I was thinking about. I just didn't check that the headers were in the database, and instead just relied on our tests of the `write_bitcoin_block` function and skipped the check (at least that's what I remember thinking when I made the commit). I don't mind adding a header check though.

Reply:
Resolving as per offline chat: the current approach seems fine and to trigger the situation I was scared of we need to mess things pretty badly in the test setup, so it was probably a non issue in the first place.